dramatiq_crontab/management/commands/crontab.py

```diff
@@ -43,16 +44,24 @@ def handle(self, *args, **options):
             importlib.import_module("dramatiq_crontab.tasks")
             self.stdout.write("Scheduling heartbeat.")
         try:
+            if not isinstance(utils.lock, utils.FakeLock):
+                self.stdout.write("Acquiring lock…")
             # Lock scheduler to prevent multiple instances from running.
-            with lock:
+            with utils.lock:
                 self.launch_scheduler()
-        except LockError as e:
+        except utils.LockError as e:
             capture_exception(e)
             self.stderr.write("Another scheduler is already running.")
 
     def launch_scheduler(self):
         signal.signal(signal.SIGTERM, kill_softly)
         self.stdout.write(self.style.SUCCESS("Starting scheduler…"))
+        scheduler.add_job(
+            utils.lock.extend,
+            IntervalTrigger(seconds=conf.get_settings().LOCK_REFRESH_INTERVAL),
+            args=(conf.get_settings().LOCK_TIMEOUT, True),
```

Comment:
I wonder whether this will work.
You are replacing the TTL of the lock with basically the same TTL. IMO: If you really want to extend the lock, you probably will need `replace_ttl=False`.

Reply:
Yes, the docs are not clear, but it's the correct behavior. I checked the redis implementation. We want "reset" the timer to another 10 seconds. Not add 10 to the remaining 5 and end up with a growing timer.